homeassistant/components/tessie/sensor.py

```diff
@@ -0,0 +1,216 @@
+        device_class=SensorDeviceClass.PRESSURE,
+        suggested_display_precision=1,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="vehicle_state-tpms_pressure_rr",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfPressure.BAR,
+        suggested_unit_of_measurement=UnitOfPressure.PSI,
+        device_class=SensorDeviceClass.PRESSURE,
+        suggested_display_precision=1,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="climate_state-inside_temp",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key="climate_state-outside_temp",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        suggested_display_precision=1,
+    ),
+    SensorEntityDescription(
+        key="climate_state-driver_temp_setting",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        suggested_display_precision=1,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    SensorEntityDescription(
+        key="climate_state-passenger_temp_setting",
+        state_class=SensorStateClass.MEASUREMENT,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        suggested_display_precision=1,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Tessie sensor platform from a config entry."""
+    coordinators = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        TessieSensorEntity(coordinator, description)
+        for coordinator in coordinators
+        for description in DESCRIPTIONS
+        if description.key in coordinator.data
+    )
+
+
+class TessieSensorEntity(TessieEntity, SensorEntity):
+    """Base class for Tessie metric sensors."""
+
+    entity_description: SensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: TessieDataUpdateCoordinator,
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, description.key)
+        self.entity_description = description
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the sensor."""
+        if isinstance(self.value, str):
+            # Make value lower case for translation
+            self.value.lower()
```

Comment:
```suggestion
            return self.value.lower()
```

Reply:
Still doesn't work though
```
ValueError: Sensor sensor.model_x_y_shift_state provides state value 'p', which is not in the list of options provided
```

Reply:
It works now. I missed return and options.